File: llm_generated_code/ChatGPT/CPP/Level1/Q20.cpp
#include <iostream>

double myPow(double x, int n) {
    double result = 1.0;
    long long power = n;
    if (power < 0) {
        x = 1 / x;
        power = -power;
    }
    while (power) {
        if (power % 2 != 0) {
            result *= x;
        }
        x *= x;
        power /= 2;
    }
    return result;
}

int main() {
    double x = 2.0;
    int n = 10;
    std::cout << "Result: " << myPow(x, n) << std::endl;
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level1/Q20.py
def power(x, n):
    if n == 0:
        return 1
    elif n < 0:
        x, n = 1 / x, -n
    result = 1
    while n:
        if n % 2:
            result *= x
        x *= x
        n //= 2
    return result
